### PR TITLE
fix(rtrace): string cache collection fix

### LINKED ISSUE #320: Strange output when executing toString() on a number
~~~
describe("example", () => {
  it("breaks", () => {
    for (let i = 0; i < 3; i++) {
      log(i.toString());
      log(i);
    }
  });
});
~~~

The above code logs the following:

~~~
✔ assembly/__tests__/example.spec.ts Pass: 1 / 1 Todo: 0 Time: 10.21ms
     [Log]: "0"
     [Log]: 0
     [Log]: "1"
     [Log]: 1
     [Log]: "1"
     [Log]: 2
~~~

In other words, the toString is generating the wrong output on the third iteration!

### PR DESCRIPTION
**Pull request by jtenner**
Closes #320 

Fascinating string caching problem involving an object overhead miscalculation. We will need to wait for an upstream feature involving exporting the RTrace OBJECT_OVERHEAD and TOTAL_OVERHEAD constants.

Depending on when the pull request pulls, I will release this fix with a small version bump.

### CODE REVIEW

File: packages/core/src/test/TestContext.ts [hunk 1]
```diff
@@ -865,7 +865,7 @@ export class TestContext {
     this.freeCount += 1;
     this.targetNode.frees += 1;
     // remove any cached strings at this pointer
-    this.cachedStrings.delete(block + 16);
+    this.cachedStrings.delete(block + 20);
```
Comment (willemneal) [inline packages/core/src/test/TestContext.ts:868]: good find!

**Review by willemneal — APPROVED** [review]
LGTM

File: packages/core/src/test/TestContext.ts (hunk 1 above)
Comment (jtenner) [reply packages/core/src/test/TestContext.ts:868]: This will become `TOTAL_OVERHEAD` when the compiler publishes a release tonight.

**Comment by jtenner** [comment]
I am going to accept this pull request despite CI not passing.

@willemneal please feel free to help fix the CI problems.

**Pull request merged**